mods/taskbar-dock-animation.wh.cpp

```diff
@@ -0,0 +1,436 @@
+        {
+            {
+                LR"(public: virtual int __cdecl winrt::impl::produce<struct winrt::Taskbar::implementation::TaskbarFrame,struct winrt::Windows::UI::Xaml::IFrameworkElementOverrides>::MeasureOverride(struct winrt::Windows::Foundation::Size,struct winrt::Windows::Foundation::Size *))"
+            },
+            &TaskbarFrame_MeasureOverride_Original,
+            TaskbarFrame_MeasureOverride_Hook,
+        },
+    };
+    if (!HookSymbols(module, taskbarViewHooks, ARRAYSIZE(taskbarViewHooks))) {
+        Wh_Log(L"DockAnimation: HookSymbols failed.");
+        return false;
+    }
+    Wh_Log(L"DockAnimation: HookSymbols succeeded for TaskbarFrame_MeasureOverride.");
+    return true;
+}
+
+using LoadLibraryExW_t = decltype(&LoadLibraryExW);
+LoadLibraryExW_t LoadLibraryExW_Original;
+
+// Lazy hook path (if Taskbar.View.dll loads later)
+HMODULE WINAPI LoadLibraryExW_Hook(LPCWSTR lpLibFileName, HANDLE hFile, DWORD dwFlags) {
+    HMODULE module = LoadLibraryExW_Original(lpLibFileName, hFile, dwFlags);
+    if (!module) return module;
+
+    if (!g_taskbarViewDllLoaded && GetTaskbarViewModuleHandle() == module) {
+        if (!g_taskbarViewDllLoaded.exchange(true)) {
+            Wh_Log(L"DockAnimation: Taskbar.View.dll loaded, hooking symbols...");
+            if (HookTaskbarViewDllSymbols(module)) {
+                Wh_ApplyHookOperations();
+                Wh_Log(L"DockAnimation: Hooks applied.");
+            }
+        }
+    }
+    return module;
+}
+
+// Windhawk entry points
+BOOL Wh_ModInit() {
+    Wh_Log(L"DockAnimation: Wh_ModInit");
+    LoadSettings();
+
+    if (HMODULE taskbarViewModule = GetTaskbarViewModuleHandle()) {
+        // Fast path (module already loaded)
+        g_taskbarViewDllLoaded = true;
+        if (!HookTaskbarViewDllSymbols(taskbarViewModule)) return FALSE;
+        g_fastPathHooksApplied = false;
+    } else {
+        // Lazy path (wait for LoadLibrary of Taskbar.View.dll)
+        HMODULE kernelBaseModule = GetModuleHandle(L"kernelbase.dll");
+        auto pKernelBaseLibraryExW = (decltype(&LoadLibraryExW))GetProcAddress(kernelBaseModule, "LoadLibraryExW");
+        WindhawkUtils::SetFunctionHook(
+            pKernelBaseLibraryExW,
+            LoadLibraryExW_Hook,
+            &LoadLibraryExW_Original
+        );
+    }
+    return TRUE;
+}
+
+void Wh_ModAfterInit() {
+    Wh_Log(L"DockAnimation: Wh_ModAfterInit");
+    if (g_taskbarViewDllLoaded && !g_fastPathHooksApplied) {
+        Wh_ApplyHookOperations();
+        g_fastPathHooksApplied = true;
+    }
+    DisableTaskbarIconClipping();
+}
+
+// Cleanup
+void Wh_ModBeforeUninit() {
+    Wh_Log(L"DockAnimation: Wh_ModBeforeUninit");
+    try {
+        for (auto& pair : g_contexts) {
+            auto& ctx = pair.second;
+            if (ctx.taskbarFrame) {
+                ctx.taskbarFrame.PointerMoved(ctx.pointerMovedToken);
+                ctx.taskbarFrame.PointerExited(ctx.pointerExitedToken);

```

Comment:
This won't work - `Wh_ModBeforeUninit` runs in a different thread than the UI thread. That's why Explorer crashes on unload. You can find the taskbar thread with `FindWindow(L"Shell_TrayWnd", NULL)` and `GetWindowThreadProcessId`, and then use `RunFromWindowThread` which can be copied from one of the mods, for example:
https://github.com/ramensoftware/windhawk-mods/blob/a44c656ddf26e242a365576bb63f8a0e5b66d36a/mods/taskbar-start-button-position.wh.cpp#L329